Added functionality to return acoustic and language model costs (#1)
EducationalTestingService/vosk-api: README.md

For installation instructions, examples and documentation visit [Vosk
Website](https://alphacephei.com/vosk).

# Test the functionality implemented in this repository using C scripts

1. Clone the `ETS fork` of the `Vosk API` repositiory [https://github.com/EducationalTestingService/vosk-api](https://github.com/EducationalTestingService/vosk-api).

2. The docker image for building Kaldi and the Python wheels is located in the cloned repository under `travis`.

   Build and run the docker image as follows:

   ```
   docker build --file Dockerfile.manylinux --tag alphacep/kaldi-manylinux:latest .
   docker run -ti -v $(pwd)/..:/io alphacep/kaldi-manylinux /bin/bash
   ```

> 3. Inside the docker container under the `src` folder run `KALDI_ROOT=/opt/kaldi make all` to compile your changes.

Reviewer: ```suggestion
3. Inside the docker container under the `io/src` folder run `KALDI_ROOT=/opt/kaldi make all` to compile your changes.
```

> 4. Go to the `c` folder and run `KALDI_ROOT=/opt/kaldi make all` to compile the C scripts. This should create an executable for the C script. To test the custom functionalities built by ETS team, run the command `./test_phone_results`. This should print out the expected results for the script.

Reviewer: ```suggestion
4. Go to the `io/c` folder and run `KALDI_ROOT=/opt/kaldi make all` to compile the C scripts. This should create an executable for the C script. To test the custom functionalities built by ETS team, run the command `./test_phone_results`. This should print out the expected results for the script.
```

Reviewer: `./test_phone_results` doesn’t work - it asks for a model. I imagine we need to download that from the vosk folks? Let’s add the instructions for that too.

Author: Yup, I've added the instructions. However, none of Vosk's models will work out of the box as they don't include the `phones.txt` file under the `model/graph` directory so I've added a `NOTE` for that as well to the `README`.

For installation instructions, examples and documentation visit [Vosk
Website](https://alphacephei.com/vosk).

# Test the functionality implemented in this repository using C scripts

1. Clone the `ETS fork` of the `Vosk API` repository [https://github.com/EducationalTestingService/vosk-api](https://github.com/EducationalTestingService/vosk-api).

> 2. If you are using your own custom ASR model, copy the model into the `model` directory in the structure shared in the section `Model structure` [on VOSK-API website](https://alphacephei.com/vosk/models). Alternatively, you can download one of the [open-source ASR models available on Vosk-API website](https://alphacephei.com/vosk/models) and copy them into the `model` directory.

Reviewer: ```suggestion
2. If you are using your own custom ASR model, copy the model into the `model` directory in the structure shared in the section `Model structure` [on VOSK-API website](https://alphacephei.com/vosk/models). Alternatively, you can download one of the [open-source ASR models available on Vosk-API website](https://alphacephei.com/vosk/models),  uncompress the downloaded zip file and, rename the containing directory to `model`. 
```

3. Add the audio file to be used to test the C test script under `c` and rename the file to `test.wav` to use the script as is. 

4. The docker image for building Kaldi and the Python wheels is located in the cloned repository under `travis`.

   Build and run the docker image as follows:

   ```
   docker build --file Dockerfile.manylinux --tag alphacep/kaldi-manylinux:latest .
   docker run -ti -v $(pwd)/..:/io alphacep/kaldi-manylinux /bin/bash
   ```

5. Inside the docker container under the `io/src` folder run `KALDI_ROOT=/opt/kaldi make all` to compile your changes.

> 6. Go to the `io/c` folder and run `KALDI_ROOT=/opt/kaldi make all` to compile the C scripts. This should create an executable for the C script. To test the custom functionalities built by ETS team, run the command `./test_phone_results`. This should print out the expected results for the script.

Reviewer: ```suggestion
6. Go to the `io/c` folder and run `ln -s ../model .` to symlink the model directory you created in step 2 inside this folder. Next, run `KALDI_ROOT=/opt/kaldi make all` to compile the C scripts. This should create an executable for the C script. To test the custom functionalities built by ETS team, run the command `./test_phone_results`. This should print out the expected results for the script.
```
